h/admin/views/badge.py

```diff
@@ -1,40 +1,53 @@
 # -*- coding: utf-8 -*-
 
+from pyramid import httpexceptions
 from pyramid.view import view_config
+from sqlalchemy.exc import IntegrityError
 
 from h import models
+from h.i18n import TranslationString as _
 
 
 @view_config(route_name='admin_badge',
              request_method='GET',
              renderer='h:templates/admin/badge.html.jinja2',
              permission='admin_badge')
-def badge_index(_):
-    return {"uris": models.Blocklist.all()}
+def badge_index(request):
+    return {"uris": request.db.query(models.Blocklist).all()}
 
 
 @view_config(route_name='admin_badge',
              request_method='POST',
              request_param='add',
-             renderer='h:templates/admin/badge.html.jinja2',
              permission='admin_badge')
 def badge_add(request):
+    uri = request.params['add']
+    item = models.Blocklist(uri=uri)
+    request.db.add(item)
```

Comment:
It's not a big deal but I wouldn't normally introduce a local variable without a reason, e.g. that it was used in multiple places, that it had a name that helped to make the code more readable, that it made it possible to break up a really long line, etc. Here I don't see what it adds, but it's not doing any harm either.